DMCompiler/DM/Builders/DMProcBuilder.cs

```diff
@@ -487,6 +487,11 @@ public void ProcessStatementFor(DMASTProcStatementFor statementFor) {
                                 outputExpr = exprRange.Value;
                             }
 
+                            if (outputExpr is DMASTIdentifier identifier &&
+                                dmObject.GetVariable(identifier.Identifier) is { IsConst: true }) {
```

Comment:
An identifier can refer to a lot more than just an object variable. This should use `outputVar` instead.

The same check should probably also be done in the `DMASTExpressionIn` case.